### PR TITLE
Feat/homepage

### PR DESCRIPTION
**Pull request by viru2001**
- Added category wise filter to videos

note : Review files in src/frontend folder and ignore all files in src/backend folder.

videos-filter (1)

### CODE REVIEW

**Review by TejasShekar — APPROVED** [review]
Code is clean. UI looked good as well. Remove all unnecessary comments and redundant code.

File: src/frontend/context/VideosContext.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const VideosContext = createContext();
+
+const VideosProvider = ({ children }) => {
+  const [videosState, videosDispatch] = useReducer(
+    videosReducer,
+    videosInitialState
+  );
+
+  //   const getVideos = async () => {
```
Comment (irajbhinde) [inline src/frontend/context/VideosContext.jsx:12]: remove commented code

File: src/backend/controllers/PlaylistController.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+/**
+ * This handler handles getting videos from user's playlist.
+ * send GET Request at /api/user/playlists/:playlistId
+ * */
+
+export const getVideosFromPlaylistHandler = function (schema, request) {
+  const user = requiresAuth.call(this, request);
+  if (user) {
+    const playlistId = request.params.playlistId;
+    const playlist = user.playlists.find((item) => item._id !== playlistId);
```
Comment (irajbhinde) [inline src/backend/controllers/PlaylistController.js:92]: make this === instead of !==

**Review by irajbhinde — COMMENTED** [review]
Looks good to merge. Just make the required changes in playlist controller

File: src/backend/controllers/PlaylistController.js (hunk 2 above)
Comment (viru2001) [reply src/backend/controllers/PlaylistController.js:92]: ok will do

**Pull request merged**